packages/hooks/use-throttle-render/index.ts

```diff
@@ -1,29 +1,60 @@
 import { onMounted, ref, watch } from 'vue'
+import type { SkeletonThrottle } from '@element-plus/components'
 
 import type { Ref } from 'vue'
 
-export const useThrottleRender = (loading: Ref<boolean>, throttle = 0) => {
+export const useThrottleRender = (
+  loading: Ref<boolean>,
+  throttle: SkeletonThrottle = 0
+) => {
   if (throttle === 0) return loading
-  const throttled = ref(false)
+  const initVal =
+    typeof throttle === 'object' ? throttle.initVal ?? false : false
+  const throttled = ref(initVal)
   let timeoutHandle: ReturnType<typeof setTimeout> | null = null
 
-  const dispatchThrottling = () => {
+  const dispatchThrottling = (timer: number) => {
     if (timeoutHandle) {
       clearTimeout(timeoutHandle)
     }
     timeoutHandle = setTimeout(() => {
       throttled.value = loading.value
-    }, throttle)
+    }, timer)
   }
-  onMounted(dispatchThrottling)
+
+  const leadingDispatch = () => {
+    if (typeof throttle === 'number') {
+      dispatchThrottling(throttle)
+    } else if (
+      typeof throttle === 'object' &&
+      throttle.leading &&
+      throttle.leading > 0
+    ) {
+      dispatchThrottling(throttle.leading)
+    }
+  }
+
+  const trailingDispatch = () => {
+    if (
+      typeof throttle === 'object' &&
+      throttle.trailing &&
+      throttle.trailing > 0
+    ) {
+      dispatchThrottling(throttle.trailing)
+    } else {
+      throttled.value = false
+    }
+  }
```

Comment:
There is a lot of overlap between these two functions, and perhaps they can be combined into one function.

Some type determination operations can directly reference internal encapsulated tool functions.
https://github.com/element-plus/element-plus/blob/72dd11aac271e788da45e0e92b3ad3dd343c1601/packages/utils/types.ts#L18

Reply:
I'm not quite clear about what you mean. The `use-throttle-render/index.ts` mainly exposes a throttle hook, and `utlils/types.ts` contains mostly pure function logic. These two can't be simply merged into one, can they? Could you please elaborate a bit more, and then I can see what I might be able to do?

Reply:
Sorry, it looks like my description was a bit confusing.

 That some of the code logic of function `leadingDispatch` and function `trailingDispatch` is the same. These two functions can be merged into one function by passing a Boolean value to distinguish the different logics.

The second sentence describes that code like `typeof throttle === 'number'` can be processed by referencing the internal encapsulated function and replaced with `isNumber(throttle)`.

Reply:
Thank you for your patient explanation, I have made a revision, please review it again.